orchagent/intfsorch.cpp

```diff
@@ -146,9 +140,16 @@ void IntfsOrch::doTask(Consumer &consumer)
                 continue;
             }
 
-            addSubnetRoute(port, ip_prefix);
             addIp2MeRoute(ip_prefix);
-            if(port.m_type == Port::VLAN && ip_prefix.isV4())
+
+            /* Add subnet route for non-loopback interfaces */
+            if (port.m_type != Port::LOOPBACK)
+            {
+                addSubnetRoute(port, ip_prefix);
```

Comment:
This condition in sonic didn't seem correct before and doesn't seem correct now either. There is no restriction that loopbacks are /32 or /128 addresses. Seems to me if we don't install the subnet route, we can cause a routing loop with a default in place.

So what's the reason for skipping addSubnetRoute for loopbacks?

Reply:
what about adding the logic of checking if it's a /32 or /128 addresses in the addSubnetRoute function. if it's a subnet address, then we add the subnet route.

Reply:
Can you explain what happens if the loopback configuration has changed? How will that be handled? At first glance it seems we won't reflect the right ip2meroute.